src/transformation/visitors/call.ts

```diff
@@ -300,6 +296,23 @@ export const transformCallExpression: FunctionVisitor<ts.CallExpression> = (node
     return wrapResultInTable ? wrapInTable(callExpression) : callExpression;
 };
 
+function isContextualCallExpression(
+    context: TransformationContext,
+    node: ts.CallExpression,
+    signature: ts.Signature | undefined
+): boolean {
+    const declaration = signature?.getDeclaration();
+    if (!declaration) {
+        if (ts.nodeIsSynthesized(node.expression)) {
+            // this includes JSX createElement calls
+            // For this we use noImplicitSelf option
+            return !context.options.noImplicitSelf;
```

Comment:
Can't we just use this return for any node without declaration? In what case does this fail?

Reply:
I just checked and it works, will change.

I somehow recall that it broke some tests, I guess that is (no longer?) the case.